components/molecule/pagination/src/index.js

```diff
@@ -127,31 +132,87 @@ const MoleculePagination = ({
-          <PageButton
-            key={pageRange}
-            page={pageRange}
-            design={
-              pageRange === page
-                ? selectedPageButtonDesign
-                : nonSelectedPageButtonDesign
-            }
-            color={
-              pageRange === page
-                ? selectedPageButtonColor
-                : nonSelectedPageButtonColor
-            }
-            onSelectPage={onSelectPage}
-            {...linkProps(pageRange)}
-          >
-            {pageRange}
-          </PageButton>
-        ))
+        <>
+          {showEdges && (
+            <>
+              <PageButton
+                key={1}
+                page={1}
+                design={
+                  page === 1
+                    ? selectedPageButtonDesign
+                    : nonSelectedPageButtonDesign
+                }
+                color={
+                  page === 1
+                    ? selectedPageButtonColor
+                    : nonSelectedPageButtonColor
+                }
+                onSelectPage={onSelectPage}
+                size={size}
+                {...linkProps(1)}
+              >
+                {1}
+              </PageButton>
+              {page > showPages && showPages + 2 < totalPages - 1 && (
+                <li className={`${BASE_CLASS}-divider`}>{divider}</li>
+              )}
+            </>
+          )}
+          {range.map(pageRange => (
+            <PageButton
+              key={pageRange}
+              page={pageRange}
+              design={
+                pageRange === page
+                  ? selectedPageButtonDesign
+                  : nonSelectedPageButtonDesign
+              }
+              color={
+                pageRange === page
+                  ? selectedPageButtonColor
+                  : nonSelectedPageButtonColor
+              }
+              onSelectPage={onSelectPage}
+              size={size}
+              {...linkProps(pageRange)}
+            >
+              {pageRange}
+            </PageButton>
+          ))}
+          {showEdges && totalPages > 1 && (
+            <>
+              {range[range.length - 1] < totalPages - 1 && (
+                <li className={`${BASE_CLASS}-divider`}>{divider}</li>
+              )}
+              <PageButton
+                key={totalPages}
+                page={totalPages}
+                design={
+                  page === totalPages
+                    ? selectedPageButtonDesign
+                    : nonSelectedPageButtonDesign
+                }
+                color={
+                  page === totalPages
+                    ? selectedPageButtonColor
+                    : nonSelectedPageButtonColor
+                }
+                onSelectPage={onSelectPage}
+                size={size}
+                {...linkProps(totalPages)}
+              >
+                {totalPages}
+              </PageButton>
+            </>
+          )}
+        </>
```

Comment:
I think that file is too large, it makes it hard to read and maintenance, do you think we could split it somehow? Anyway this is something that happens in a lot of other components so I'm fine to merge it like this, but I think would be nice to think ways to split components this large

Reply:
I don't think that's necessary. We could create a component for the edges but I don't think that makes sense because it's only going to be used for pagination.